packages/web-app-files/src/views/spaces/DriveResolver.vue

```diff
@@ -28,11 +37,39 @@ export default defineComponent({
   },
   setup() {
     const store = useStore()
+    const clientService = useClientService()
+    const router = useRouter()
     const driveAliasAndItem = useRouteParam('driveAliasAndItem')
     const isSpaceRoute = useActiveLocation(isLocationSpacesActive, 'files-spaces-generic')
     const isTrashRoute = useActiveLocation(isLocationTrashActive, 'files-trash-generic')
+    const resolvedDrive = useDriveResolver({ store, driveAliasAndItem })
+
+    const getSpaceResource = async (): Promise<SpaceResource> => {
+      const space = unref(resolvedDrive.space)
+      try {
+        const publicSpace = (await clientService.webdav.getFileInfo(space)) as SpaceResource
+        return publicSpace
+      } catch (e) {
+        return space
+      }
+    }
+
+    onMounted(async () => {
+      const space = unref(resolvedDrive.space)
+      if (space && isPublicSpaceResource(space)) {
+        let publicSpace = (await getSpaceResource()) as PublicSpaceResource
```

Comment:
What happens if this request fails for whatever reason?

Reply:
The space from `useDriveResolver()` will be used as default. It won't have `publicLinkPermission` though, so the following if-condition will never resolve to `true`.

Reply:
Yeah, my point is: should we show some kind of error?

Reply:
Hmm IMO we don't need a user facing error here, as a user would most likely not know what to do with it anyway. But a console error would be good indeed. I'll wait for the review from Benedikt and add it after.

Reply:
Hm... I think it would make more sense to do the `getFileInfo` before inserting the `PublicSpaceResource` into the store, i.e. inside the `store.watch` of the `web-runtime/src/index.ts`.

Reply:
for the record: this was a can of worms. 😅